server/lib/adapters/mssql.js

```diff
@@ -16,12 +16,14 @@ import validateResultColumns from "./validate-result-columns";
 export function parseConnectionConfig(settings) {
   const conn = ["type", "host", "port", "name", "user", "password"].reduce((c, key) => {
     const val = settings[`db_${key}`];
-    if (c && val && val.length > 0) {
-      return {
-        ...c,
-        [key]: val
-      };
+    // if (c && val && val.length > 0) {
+    //   return { ...c, [key]: val };
+    // }
+    if (c && val && (val.length > 0 || (key === "port" && _.isNumber(val)))) {
```

Comment:
previously when port is a number, this if evaluates to false, and then we return false, which on a "reduce" call will return false given the current logic